#9392 Added possibility to print qr code

MrTschi opened the pull request.

matiu commented:
thanks a lot for the contribution! we will review it soon!

Gamboster commented on src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.html:
```diff
@@ -84,10 +84,20 @@
         <ion-item-divider>
           <span translate>From the destination device, go to Add wallet &gt; Import wallet and scan this QR code</span>
         </ion-item-divider>
+        <ion-item-divider class="on-print-hide">
```
You could add *ngIf="!isCordova" here.

MrTschi commented:
Thank you for the feedback guys. I‘ll have a look into within the next days if I‘ll find a solution for android. 
If not, i‘ll add the limitation suggested by @Gamboster

MrTschi commented:
I committed a preliminary state of my work, will finalize it on weekend. Could you please give me feedback to the pullrequest?
I‘m not sure regarding the document.getElementById implementation, is there a way to do it better?

Tested it in browser and on iOS, both are working. 

![af4aa6ed-1df0-4b4a-9f7a-b10389956334](https://user-images.githubusercontent.com/9491895/49965476-9395c280-ff1d-11e8-916c-c1c707d735e9.jpeg)

Gamboster commented on src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.ts:
```diff
@@ -363,6 +372,16 @@ export class WalletExportPage extends WalletTabsChild {
 
+  public printQr(): boolean {
+    if (this.isCordova) {
+      let printContent = document.getElementById('cordovaPrintWrapper');
```
It should be `const` instead `let`

Gamboster commented on src/providers/index.ts:
```diff
@@ -1,6 +1,7 @@
 /* Native modules */
 export { AndroidFingerprintAuth } from '@ionic-native/android-fingerprint-auth';
 export { Clipboard } from '@ionic-native/clipboard';
+export { Printer } from '@ionic-native/printer';
```
Please, in alphabetical order

Gamboster commented on package-lock.json:
```diff
@@ -557,6 +557,11 @@
       "resolved": "https://registry.npmjs.org/@ionic-native/launch-review/-/launch-review-4.14.0.tgz",
       "integrity": "sha512-8Yl0r+RmEoaCulccUvLjWDsTw2MV2dKMLccvcOfzOQv2iJnnnBMQ43JsFOxX02ufRc4V5OafPqYGNAN+Krxisw=="
     },
+    "@ionic-native/printer": {
```
Do not commit the package-lock file

Gamboster commented on package.json:
```diff
@@ -167,6 +167,7 @@
     "@ionic-native/fcm": "4.14.0",
     "@ionic-native/file": "4.14.0",
     "@ionic-native/launch-review": "4.14.0",
+    "@ionic-native/printer": "4.18.0",
```
Use the same version that we are using for other plugins, please (4.14.0)

Gamboster commented on src/pages/settings/wallet-settings/wallet-settings-advanced/wallet-export/wallet-export.scss:
```diff
@@ -7,9 +7,23 @@ page-wallet-export {
     margin-top: 5rem;
     text-align: center;
   }
+  @media print {
```
Try to center the image and text when print in mobile

MrTschi replied:
The CSS is only available when printing on desktop. Within cordova, only the div element <div id="cordovaPrintWrapper"> is passed to the printer. If I'm going to pass the whole <html> or <body>, the printing preview on the device is showing an empty page.

Gamboster commented:
I addition, you could add an option to `printer.print(content, options)` function, and set a name for the file generated. Maybe something like this:
```let options: PrintOptions = {
     name: 'Exported wallet from ' + appName
   };```
To prevent the "unknown" name.

Also, you can add some logger, in `isAvailable().then()`

Gamboster commented:
@MrTschi 
As a security policy, we can not merge commits that have not been signed.
Could you do that last thing? It even seems that you have sent commits from different users
![_9392_added_possibility_to_print_qr_code_by_mrtschi_ _pull_request__9393_ _bitpay_copay](https://user-images.githubusercontent.com/10983458/50094187-59644380-01f1-11e9-839d-e8478497b30a.jpg)

MrTschi commented:
Hi guys, thanks for the feedback!
I rebased my commits to sign them and to updated commit author to the correct one:
```
git commit -S --amend --author "MrTschi <______@icloud.com>" --no-edit && git rebase --continue
```
Finally I force-pushed it.
I also updated the CONTRIBUTION.md and added the fact that a commit must be signed.
